appservice/src/tree/DeploymentTreeItem.ts

```diff
@@ -144,6 +148,30 @@ export class DeploymentTreeItem extends AzureTreeItem<ISiteTreeRoot> {
         });
     }
 
+    public async viewCommitInGitHub(): Promise<void> {
+        // if it's not connected to a .git repo, this won't work
```

Comment:
So you're doing quite a few checks here after the user clicks "View Commit in GitHub", but I think this is a case where I would rather just hide the context menu preemptively (if `ScmType !== Github`). I think we've gone to far down the rabbit hole of "never hide a context menu". I think it's okay in situations where the UI makes clear that nodes are different types or in different states. For example, the Cosmos DB extension has "Mongo" and "SQL" nodes at the same level with different context menus, but it's fine since we know they're different types. Just make clear in the UI if I'm connected to Github, Local Git, Zip deploy, etc.